pkg/loop/internal/pb/chain_writer.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+option go_package = "github.com/smartcontractkit/chainlink-common/pkg/loop/internal/pb";
+
+package loop;
+
+import "codec.proto";
+import "relayer.proto";
+import "google/protobuf/empty.proto";
+
+service ChainWriter {
+    rpc SubmitTransaction(SubmitTransactionRequest) returns (google.protobuf.Empty) {}
+    rpc GetTransactionStatus(GetTransactionStatusRequest) returns (GetTransactionStatusReply) {}
+    rpc GetFeeComponents(google.protobuf.Empty) returns (GetFeeComponentsReply) {}
+}
+
+message SubmitTransactionRequest {
+    string contract_name = 1;
+    string method = 2;
+    VersionedBytes params = 3;
+    string transaction_id = 4;
+    string to_address = 5;
+    TransactionMeta meta = 6;
+    BigInt value = 7;
+}
+
+message TransactionMeta {
```

Comment:
why is this it's own message type?

if it's going to be extended and generic, did you using a map key-value pairs, like an http header and letting the application interpret as needed?

if it's not going to be extended and generic, why not embed the wf id field in the request

Reply:
This is just a proto representation of the `TxMeta` struct that the `ChainWriter` uses. It might very well be extended over time. If we use KV pairs we'll probably have to use reflection to map the keys to struct fields, which seems like more work than maintaining the message.